tests/test_TFUtil.py

```diff
@@ -293,6 +293,65 @@ def test_Data_copy_compatible_to_time_major():
   assert d2a.feature_dim_axis == d1.feature_dim_axis
 
 
+def test_Data_find_matching_dim_map_different_static_dims():
+  d1 = Data(name='p1_output', shape=(5, 5, 3), batch_dim_axis=None, time_dim_axis=None)  # [5,5,F|3]
+  d2 = Data(name='p2_output', shape=(5, 1, 1), batch_dim_axis=None, time_dim_axis=None)  # [5,1,F|1]
+
+  # default should not match
+  is_equal_opts = dict(allow_same_feature_dim=True, allow_same_spatial_dim=True, treat_feature_as_spatial=True)
+  assert_equal(d1.find_matching_dims(d2.get_dim_tag(0), is_equal_opts=is_equal_opts), [0,1])
+  assert_equal(d1.find_matching_dims(d2.get_dim_tag(1), is_equal_opts=is_equal_opts), [])
+  assert_equal(d1.find_matching_dims(d2.get_dim_tag(2), is_equal_opts=is_equal_opts), [])
+
+  failed = False
+  try:
```

Comment:
You could use `nose.tools.assert_raises` here.

Reply:
Oh, nevermind, this is only an intermediate commit.

Reply:
Thanks anyway, I didn't know that existed. Good to know definitely!